new/chef-workstation.md

```diff
@@ -0,0 +1,108 @@
+---
+RFC: unassigned
+Title: The Future of Chef Workstation
+Author:
+- Jon Morrow <jmorrow@chef.io>
+- Noah Kantrowitz <noah@coderanger.net>
+Status: Draft
+Type: Informational
+---
+
+# The Future of Chef Workstation
+
+ChefConf 2018 saw the unveiling of both the `chef-run` command and a Chef Workstation
+installer package containing it. This has lead to user confusion on two fronts.
+First, the term "Chef run" (or "Chef client run") is already in very common use
+in our community, so trying to communicate about `chef-run` has been tricky.
+Second, having two installers (Chef Workstation and ChefDK) has resulted in a
+lack of certainty about which should be used.
+
+## Motivation
+
+    As a Chef user,
+    I want to use the new Chef functionality,
+    so that I can manage machines/resources without Chef Server.
+
+    As a Chef community member,
+    I want to communicate with other Chef community members,
+    so that I can share knowledge and answer questions.
+
+## Specification
+
+1. The `chef-run` command will be renamed to `chef-target` [ed: not attached to this name, suggest a better on in comments].
+2. The `chef-target` gem will be split into its own repository for future development.
+3. The installer currently called "ChefDK" will be renamed "Chef Workstation".
+   This will require setting up documentation to explain the rename, as well as
+   redirects and supporting pages in the download system to ensure users don't
+   get lost.
+4. The `chef-dk` gem will not be renamed at this time, as few users see the gem
```

Comment:
I'm sensing a common theme that we may want to address as part of this RFR (or another RFR). Basically: Is Chef Workstation / ChefDK an actual tool with its own functionality plus a bunch of packaged components? Or is it just the installer that bundles a bunch of components, one of those components being a `chef` command?

I think I prefer the latter. Thoughts?

Reply:
I tend to agree with the idea of an installer that bundles components as a longer term vision. One caveat might be components such as a tray app and/or management cli for the package itself.

Reply:
This is why I tried to draw a sharp distinction between "the tool" and "the installer", but that said, we should still have one primary entry point for new Chef (the project) users, and if that isn't going to be the `chef` CLI, that probably needs some discussion and a lot of justification (since `chef` was already our "do over" from `knife`).

Reply:
I agree with this sharp distinction. Here's my understanding of the "just the installer" option for what Workstation/DK becomes:

* the "product"/project under which end-user tooling is available
* the release artifact (installer) for that tooling
* directly responsible for the source code of "entry point" utilities (e.g. a tray app or CLI command suite root like `chef`)

Actual functionality would be implemented in component tools that the `chef` root command dispatches to, as the run/target/ad-hoc capability is described in this RFC. Some of our current muddy waters come from that the `chef` command—and all of its sub-command logic for policy files and such—is currently implemented in the `chef-dk` gem.